modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala

```diff
@@ -2457,4 +2469,34 @@ abstract class RunTestDefinitions
         processes.foreach { case (p, _) => expect(p.exitCode() == 0) }
       }
     }
+
+  def utf8ScriptContents(classname: String, fileName: String, message: String): String =
+    """
+object MAINCLASS {
+  def props(s: String): String = Option(sys.props(s)).getOrElse("")
+  val fileName = "FILENAME"
+  val utfPropnames = Seq("file.encoding", "sun.jnu.encoding", "native.encoding", "java.runtime.version", "java.home", "java.vendor.version", "java.vm.version" )
+  utfPropnames.foreach { (str: String) => System.err.println(s"$str = ${props(str)}") }
+  if (sys.props("os.name").toLowerCase.contains("windows")) {
+    val scalaVersion = util.Properties.versionString.replaceFirst("version ", "").trim
+    System.err.println(s"scala version[$scalaVersion]")
+    import scala.sys.process._
+    System.err.println(s"chcp.com code-page: ${"chcp.com".!!.trim}")
+    if (!scalaVersion.startsWith("2.12")) {
+      try {
+        val codepage = "reg query 'HKLM\\SYSTEM\\CurrentControlSet\\Control\\Nls\\CodePage' /v ACP".!!.trim
+        System.err.println(s"registry code-page: ${codepage.replaceAll("[^0-9]", "")}")
+      } catch {
+        case _: Throwable =>
+      }
+    }
+  }
+  import java.nio.charset.Charset
+  System.err.println(s"Charset.defaultCharset: ${Charset.defaultCharset}")
+  System.err.println(s"fileName Chars: ${fileName.map(c => f"U+$c%04x").mkString(" ")}")
+  def main(args: Array[String]): Unit = {
+    print("""" + message + """") // no newline needed here
+  }
+}
+""".trim.replaceAll("MAINCLASS", classname).replaceAll("FILENAME", fileName)
```

Comment:
This is... overkill. 😧 
I mean, don't get me wrong, I can see this works, but we used to just
```scala
object TestÅÄÖåäö {
  def main(args: Array[String]): Unit = {
  println("$message")
  }
}
```
😐